aeron-driver/src/main/c/aeron_publication_image.c

```diff
@@ -921,10 +921,9 @@ void aeron_publication_image_on_time_event(
                 image->conductor_fields.state = AERON_PUBLICATION_IMAGE_STATE_LINGER;
                 image->conductor_fields.time_of_last_state_change_ns = now_ns;
 
-                aeron_driver_conductor_image_transition_to_linger(conductor, image);
-
                 aeron_receive_channel_endpoint_dec_image_ref_count(image->endpoint);
                 aeron_driver_receiver_proxy_on_remove_publication_image(conductor->context->receiver_proxy, image);
+                aeron_driver_conductor_image_transition_to_linger(conductor, image);
 
```

Comment:
On another note, I think COOL_DOWN is misleading now.
Before it made sense as it was in place for the entire DRAINING period, now its there either for a split second, or for the lifetime of the subscription set for this stream

Reply:
I think we might be able to eliminate COOL_DOWN. The original intent was to stop the image from flapping. However, the nature of a slow receiver that falls outside the window has a natural resonance and won't flap anyway. So, it's worth examining what the logic would look like refactored. But that is outside the scope of this PR.